src/app.js

```diff
@@ -1,21 +1,33 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const http = require('http');
+const url = require('url');
+const { createPathArray, createParamsObj } = require('./paramsHelper');
+
+const server = http.createServer((req, res) => {
+  const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+
+  try {
+    const pathname = normalizedUrl.pathname;
+
+    if (pathname !== '/favicon.ico') {
+      const parts = createPathArray(pathname);
+      const query = createParamsObj(normalizedUrl.searchParams);
+
+      res.setHeader('Content-Type', 'application/json');
+
+      const resultObject = JSON.stringify({
+        parts,
+        query,
+      });
+
+      res.end(resultObject);
+    }
+  } catch (error) {
+    // Some error handling
```

Comment:
Error handling block is empty. It's important to handle the error properly, for example, logging the error and sending a response with the error status to the client.